ruby/lib/minitest/queue.rb

```diff
@@ -237,16 +254,24 @@ def run_from_queue(reporter, *)
         end
       end
     end
+
+    private
+
+    def current_timestamp
+      Time.now.to_i
+    end
```

Comment:
Not following how to incorporate this approach since we want the timestamps on the `MiniTest::Result` instance returned by [run](https://github.com/Shopify/ci-queue/blob/bb537df0340561eef8725a022e836e3e8b1006a0/ruby/lib/minitest/queue.rb#L162-L164). Unless we overrode `run_one_method`?

I have a proposed alternative in the latest commit d770afb7fa35b2455110ab9134295f8985ae09e8 but let me know if I'm missing anything about your original suggestion.

Reply:
Ah right, I see. Just wondering if setting the timestamps needs to go in an ensure in case of an exception. But guess we wouldn't have a result in this case either?

If you want to clean the method up you could still move it to a function and or class method in `WithTimestamps` but I will leave that up to you. No strong feelings.

```ruby
def with_timestamps
  start = Time.now.to
  result = yield
  result
ensure
  result.start_timestamp = start
  result.finish_timestamp = Time.now.to_i
end

def run
  with_timestamps do
    run_one_method
  end
end
```

```ruby
module WithTimestamps
  def self.measure
    start = Time.now.to
    result = yield
    result
  ensure
    result.start_timestamp = start
    result.finish_timestamp = Time.now.to_i
  end

def run
  WithTimestamps.measure do
    run_one_method
  end
end
```

Reply:
Went with the first one.